var/spack/repos/builtin/packages/dsfmt/targets.patch

```diff
@@ -17,7 +17,7 @@ index 5cc4ed6..a9dcd99 100644
  	rm -f *.o *~ test-*-M*
 +
 +library: dSFMT.c
-+	$(CC) $(CCFLAGS) -shared -fPIC dSFMT.c -o libdSFMT.so
++	$(CC) $(CCFLAGS) -shared -fPIC -DNDEBUG -DDSFMT_MEXP=19937 -DDSFMT_DO_NOT_USE_OLD_NAMES -DDSFMT_SHLIB dSFMT.c -o libdSFMT.so
```

Comment:
Sorry if this is obvious but would this command not overwrite the previous command? Both appear to be generating a `libdSFMT.so`

Reply:
I think the confusing bit is this is showing a meta-diff :D. But thanks for paying attention, cause I realize there are multiple issues :grimacing: : hard-coded `.so` suffix, install always running the `library` target again, and me getting the phase name wrong in spack (`make` should be `build`). On the last commit I pushed:

```
==> dsfmt: Executing phase: 'edit'
==> [2021-11-24-21:49:27.505348] Using default implementation: skipping edit phase.
==> dsfmt: Executing phase: 'build'
==> [2021-11-24-21:49:27.510644] 'make' '-j16' 'build-library'
gcc -O3 -finline-functions -fomit-frame-pointer -DNDEBUG -fno-strict-aliasing --param max-inline-insns-single=1800 -Wmissing-prototypes -Wall  -std=c99 -shared -fPIC -DNDEBUG -DDSFMT_MEXP=19937 -DDSFMT_DO_NOT_USE_OLD_NAMES -DDSFMT_SHLIB dSFMT.c -o libdSFMT.so
echo 1 > build-library
==> dsfmt: Executing phase: 'install'
==> [2021-11-24-21:49:27.687519] 'make' '-j16' 'PREFIX=/home/user/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-10.3.0/dsfmt-2.2.5-xhwb6nh5bdciho7fhbpizdrrp4pt27zl' 'install'
install -d /home/user/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-10.3.0/dsfmt-2.2.5-xhwb6nh5bdciho7fhbpizdrrp4pt27zl/lib/
install -m 644 libdSFMT.so /home/user/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-10.3.0/dsfmt-2.2.5-xhwb6nh5bdciho7fhbpizdrrp4pt27zl/lib/
==> dsfmt: Successfully installed dsfmt-2.2.5-xhwb6nh5bdciho7fhbpizdrrp4pt27zl
  Fetch: 0.01s.  Build: 0.27s.  Total: 0.27s.
```

Reply:
So, I think all issues should be gone.